src/crimson/osd/pg.cc

```diff
@@ -905,32 +931,39 @@ PG::do_osd_ops_execute(
       std::move(all_completed_fut)
     );
   }, OpsExecuter::osd_op_errorator::all_same_way(
-    [rollbacker, failure_func_ptr]
+    [this, error_func_ptr, rollbacker, failure_func_ptr]
     (const std::error_code& e) mutable {
 
-    auto submitted_fut = seastar::now();
+    PG::interruptible_future<> maybe_rollback_fut = seastar::now();
+    ceph_tid_t rep_tid = shard_services.get_tid();
 
-    auto all_completed_fut = e.value() == ENOENT ?
-      (*failure_func_ptr)(e) :
-      rollbacker.rollback_obc_if_modified(e).then_interruptible(
-      [e, failure_func_ptr] {
-          return (*failure_func_ptr)(e);
-      });
+    if (e.value() == ENOENT) {
+      maybe_rollback_fut = rollbacker.rollback_obc_if_modified(e);
+    }
 
-    return PG::do_osd_ops_iertr::make_ready_future<pg_rep_op_fut_t<Ret>>(
-      std::move(submitted_fut),
-      std::move(all_completed_fut)
-    );
+    return maybe_rollback_fut.then_interruptible(
+    [error_func_ptr, e, rep_tid, failure_func_ptr] {
+      // record error log
+      return (*error_func_ptr)(e, rep_tid).then(
```

Comment:
This appears to be the only actual user of error_func_ptr -- why not simply move the declaration of maybe_submit_error_log here and avoid needing error_func_ptr entirely?